go/adbc/README.md

```diff
@@ -0,0 +1,251 @@
+            col0.ValueStr(i), col1.ValueStr(i), col2.ValueStr(i))
+    }
+}
+```
+
+### Parameterized Queries
+
+We can bind parameters in our queries using Arrow records:
+
+```go
+import (
+    "strings"
+    "github.com/apache/arrow-go/v18/arrow"
+    "github.com/apache/arrow-go/v18/arrow/array"
+    "github.com/apache/arrow-go/v18/arrow/memory"
+)
+
+// Create parameter schema and data
+paramSchema := arrow.NewSchema([]arrow.Field{
+    {Name: "param1", Type: arrow.PrimitiveTypes.Int64},
+}, nil)
+
+params, _, err := array.RecordFromJSON(memory.DefaultAllocator, paramSchema,
+    strings.NewReader(`[{"param1": 41}]`))
+if err != nil {
+    return err
+}
+defer params.Release()
+
+// Set query and bind parameters
+err = stmt.SetSqlQuery("SELECT ? + 1 AS the_answer")
+if err != nil {
+    return err
+}
+
+err = stmt.Prepare(ctx)
+if err != nil {
+    return err
+}
+
+err = stmt.Bind(ctx, params)
+if err != nil {
+    return err
+}
+
+reader, _, err := stmt.ExecuteQuery(ctx)
+if err != nil {
+    return err
+}
+defer reader.Release()
+```
+
+### Ingesting Bulk Data
+
+The Go ADBC APIs offer methods for bulk data ingestion. We can insert Arrow data
+into a new database table:
+
+```go
+// Create Arrow table data
+schema := arrow.NewSchema([]arrow.Field{
+    {Name: "ints", Type: arrow.PrimitiveTypes.Int64, Nullable: true},
+    {Name: "strs", Type: arrow.BinaryTypes.String, Nullable: true},
+}, nil)
+
+table, _, err := array.RecordFromJSON(memory.DefaultAllocator, schema,
+    strings.NewReader(`[{"ints": 1, "strs": "a"}, {"ints": 2, "strs": null}]`))
+if err != nil {
+    return err
+}
+defer table.Release()
+
+reader, err := array.NewRecordReader(schema, []arrow.Record{table})
+if err != nil {
+    return err
+}
+defer reader.Release()
+
+// Ingest the data
+count, err := adbc.IngestStream(ctx, conn, reader, "sample",
+    adbc.OptionValueIngestModeCreateAppend, adbc.IngestStreamOptions{})
+if err != nil {
+    return err
+}
+fmt.Printf("Ingested %d rows\n", count)
+
+// Verify the data
+stmt2, err := conn.NewStatement()
+if err != nil {
+    return err
+}
+defer stmt2.Close()
+
+err = stmt2.SetSqlQuery("SELECT COUNT(DISTINCT ints) FROM sample")
+if err != nil {
+    return err
+}
+
+reader2, _, err := stmt2.ExecuteQuery(ctx)
+if err != nil {
+    return err
+}
+defer reader2.Release()
+```
+
+### Getting Database/Driver Metadata
+
+We can get information about the driver and the database:
+
+```go
+// Get driver info
+infoReader, err := conn.GetInfo(ctx, []adbc.InfoCode{
+    adbc.InfoVendorName,
+    adbc.InfoDriverName,
+})
+if err != nil {
+    return err
+}
+defer infoReader.Release()
+
+// Process the info results...
+for infoReader.Next() {
+    record := infoReader.Record()
+    // Extract vendor name, driver name, etc. from the record
+}
+```
+
+We can also query for tables and columns in the database:
+
+```go
+objectsReader, err := conn.GetObjects(ctx, adbc.ObjectDepthAll, nil, nil, nil, nil, nil)
```

Comment:
should we explain what those parameters are for?

Reply:
Yes! Done in baa25e3037051885a001c6134570779ba70f06cd.